### ISSUE TITLE
Add some instrumentation to `SessionPool`

### ISSUE THREAD

**Issue by mr-salty** [googleapis/google-cloud-cpp#4027]
There's not much visibility into what happens inside `SessionPool`, it would be nice to have some instrumentation there.  OpenCensus/OpenTelemetry seems like the right solution longer term, but perhaps we can add something relatively quickly to help with debugging (`GCP_LOG` or something?) It would be nice to see:

* When operations fail
* When we've hit the size limit on the session pool (especially when `Allocate()` has to block for a long time)
* Some indication of activity (session creation, deletion, refresh)
* Maybe some statistics (current size, total create/delete ops, blocks in Allocate waiting for a session, etc.)

**Comment by coryan** [comment]
Punting until integration with OpenTelemetry is a thing.

**Comment by coryan** [comment]
Still waiting on OpenTelemetry

**Comment by coryan** [comment]
Close, we will reopen when we revisit metrics.
